### ISSUE TITLE
Redundant property names should be omitted in anonymous classes

### ISSUE THREAD

**Issue by tvsonar** [SonarSource-VisualStudio/sonarlint-roslyn-analyzer#678]
R# _RedundantAnonymousTypePropertyName_

When constructing anonymous types, the property names can be automatically defined by the compiler if the property initialization happens through an identifier. 

```C#
var X = 5;
var Y = "my string";

var anon = new 
{
  X = X, //Noncomplaint, the new object would have the same property without the "X =" part.
  Y = Y  //Noncomplaint
};
```

we should probably not report on the following:
```C#
var X = 5;

var anon = new 
{
  X = X, //Noncomplaint, but we should not report on it, because it doesn't look good when some property names are automatically derived and some are not.
  Y = "some string"
};
```

**Comment by ganncamp** [comment]
in X=X, it's the "X=" that can be left out, not the "=X"

**Comment by ganncamp** [comment]
https://jira.sonarsource.com/browse/RSPEC-3441
